src/components/ContactInfo/ContactInfo.vue

```diff
@@ -0,0 +1,365 @@
+        <label class="d-flex flex-wrap">
+          <span class="subtitle mr-2">Phone Number</span>
+        </label>
+        <div id="lbl-phone" v-if="!!contactInfo.phone" class="info-text">
+          {{ contactInfo.phone }}
+          <span v-if="!!contactInfo.extension">Ext: {{ contactInfo.extension }}</span>
+        </div>
+        <div id="lbl-no-phone" v-else class="info-text">(Not entered)</div>
+      </v-flex>
+
+      <!-- Edit Actions -->
+      <v-flex xs1 class="mt-n2">
+        <div class="actions mr-4">
+          <v-btn
+            v-if="hasBusinessContactInfoChange"
+            id="btn-undo-contact-info"
+            text color="primary"
+            @click="resetContactInfo()"
+          >
+            <v-icon small>mdi-undo</v-icon>
+            <span>Undo</span>
+          </v-btn>
+          <v-btn
+            v-else
+            id="btn-correct-contact-info"
+            text color="primary"
+            @click="isEditing = true"
+          >
+            <v-icon small>mdi-pencil</v-icon>
+            <span>{{editLabel}}</span>
+          </v-btn>
+
+          <!-- Drop Down Actions -->
+          <span class="more-actions" v-if="hasBusinessContactInfoChange">
+            <v-menu
+              offset-y left nudge-bottom="4"
+              v-model="dropdown"
+            >
+              <template v-slot:activator="{ on }">
+                <v-btn
+                  text small color="primary"
+                  id="btn-more-actions"
+                  v-on="on"
+                >
+                  <v-icon>{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+                </v-btn>
+              </template>
+              <v-list>
+                <v-list-item
+                  id="btn-more-actions-edit"
+                  class="v-list-item"
+                  @click="isEditing = true; dropdown = false"
+                >
+                  <v-list-item-subtitle>
+                    <v-icon small color="primary">mdi-pencil</v-icon>
+                    <span class="drop-down-action ml-1">Change</span>
+                  </v-list-item-subtitle>
+                </v-list-item>
+              </v-list>
+            </v-menu>
+          </span>
+        </div>
+      </v-flex>
+    </v-layout>
+
+    <!-- Edit Contact Info -->
+    <template v-else id="edit-contact-form">
+      <v-layout row class="mx-0">
+        <v-flex xs3>
+          <label>Registered Office <br>Contact Information</label>
+        </v-flex>
+      </v-layout>
+
+      <v-layout row class="mx-0">
+        <v-flex xs3></v-flex>
+        <v-flex xs9 class="my-4 info-text">
+          There is no fee to change Registered Office Contact Information. Any changes made will be applied immediately.
+        </v-flex>
+      </v-layout>
+
+      <v-form
+        v-model="formValid"
+        ref="editContactForm"
+        name="business-contact-form"
+        class="business-contact-form pt-5"
+        @submit.prevent="submitContact()"
```

Comment:
What exactly does this do?

Reply:
Oh, does it catch the submit event from the Done button?

Reply:
Absolutely!
Catches the default submit behaviour and calls the method we define instead.